app/controllers/reports/match_progress_controller.rb

```diff
@@ -85,20 +85,26 @@ def step_names(sub_program_id)
 
     def sub_programs
       @sub_programs ||= SubProgram.
-        joins(:program).
-        preload(:program).
-        pluck(p_t[:name], sp_t[:name], :id).
-        sort
+        joins(:program, :match_route).
+        preload(:program, :match_route).
+        order(p_t[:name].asc, sp_t[:name].asc, id: :asc).
+        map do |sp|
+          {
+            program: sp.program.name,
+            sub_program: sp.name,
+            route: sp.match_route.title,
+            id: sp.id,
+          }
+        end
     end
-    helper_method :sub_programs
 
     def sub_program_list
-      @sub_program_list ||= sub_programs.map do |project_name, sub_project_name, id|
+      @sub_program_list ||= sub_programs.map do |sp|
         [
-          [project_name, sub_project_name].join('|'),
-          id,
+          [sp[:program], sp[:sub_program], sp[:route]].compact_blank.join('|'),
```

Comment:
It looks like using compact here introduces ambiguity around which names are for which type of entity. Maybe that doesn't matter unless something is parsing them out (for example this would be incorrect `program, sub_program, route = name.split('|')`)

Reply:
This is just used for display, I'd rather turn this into a sentence or more descriptive text, but didn't want to mess with the wording too much until we confirm this works.